{"name": "Random Scenario three", "status": "failed", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[text()='Documentation']\"}\n  (Session info: chrome=119.0.6045.10); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n#0 0x555555ce7ea3 <unknown>\n#1 0x5555559aaf97 <unknown>\n#2 0x5555559f8d53 <unknown>\n#3 0x5555559f8e51 <unknown>\n#4 0x555555a3e7f4 <unknown>\n#5 0x555555a1fd1d <unknown>\n#6 0x555555a3b96d <unknown>\n#7 0x555555a1fac3 <unknown>\n#8 0x5555559eaee4 <unknown>\n#9 0x5555559ebeae <unknown>\n#10 0x555555cae701 <unknown>\n#11 0x555555cb259e <unknown>\n#12 0x555555c9c635 <unknown>\n#13 0x555555cb31f6 <unknown>\n#14 0x555555c7ff0f <unknown>\n#15 0x555555cd5fe8 <unknown>\n#16 0x555555cd61b3 <unknown>\n#17 0x555555ce7033 <unknown>\n#18 0x7ffffffa6609 start_thread", "trace": ".0 = <list_iterator object at 0x119111370>\n\n>   kwargs = {arg: kwargs[arg] if arg in kwargs else request.getfixturevalue(arg) for arg in args}\n\n../venv/lib/python3.9/site-packages/pytest_bdd/scenario.py:154: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n../venv/lib/python3.9/site-packages/_pytest/fixtures.py:585: in getfixturevalue\n    fixturedef = self._get_active_fixturedef(argname)\n../venv/lib/python3.9/site-packages/_pytest/fixtures.py:607: in _get_active_fixturedef\n    self._compute_fixture_value(fixturedef)\n../venv/lib/python3.9/site-packages/_pytest/fixtures.py:693: in _compute_fixture_value\n    fixturedef.execute(request=subrequest)\n../venv/lib/python3.9/site-packages/_pytest/fixtures.py:1069: in execute\n    result = ihook.pytest_fixture_setup(fixturedef=self, request=request)\n../venv/lib/python3.9/site-packages/pluggy/_hooks.py:493: in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n../venv/lib/python3.9/site-packages/pluggy/_manager.py:115: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n../venv/lib/python3.9/site-packages/_pytest/fixtures.py:1123: in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n../venv/lib/python3.9/site-packages/_pytest/fixtures.py:902: in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n../conftest.py:130: in navigate_to_dashboards_page\n    dashboards_page = DashboardsPage()\n../src/genesis/pages/dashboard.py:9: in __init__\n    BaseClass.is_page_loaded(By.XPATH, self.__documentation_locator)\n../src/base/base_class.py:345: in is_page_loaded\n    raise error\n../src/base/base_class.py:336: in is_page_loaded\n    element = Context.driver.find_element(by, web_element)\n../venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py:738: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n../venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py:344: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x119111b50>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...wn>\\\\n#16 0x555555cd61b3 \\\\u003Cunknown>\\\\n#17 0x555555ce7033 \\\\u003Cunknown>\\\\n#18 0x7ffffffa6609 start_thread\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[text()='Documentation']\"}\nE         (Session info: chrome=119.0.6045.10); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x555555ce7ea3 <unknown>\nE       #1 0x5555559aaf97 <unknown>\nE       #2 0x5555559f8d53 <unknown>\nE       #3 0x5555559f8e51 <unknown>\nE       #4 0x555555a3e7f4 <unknown>\nE       #5 0x555555a1fd1d <unknown>\nE       #6 0x555555a3b96d <unknown>\nE       #7 0x555555a1fac3 <unknown>\nE       #8 0x5555559eaee4 <unknown>\nE       #9 0x5555559ebeae <unknown>\nE       #10 0x555555cae701 <unknown>\nE       #11 0x555555cb259e <unknown>\nE       #12 0x555555c9c635 <unknown>\nE       #13 0x555555cb31f6 <unknown>\nE       #14 0x555555c7ff0f <unknown>\nE       #15 0x555555cd5fe8 <unknown>\nE       #16 0x555555cd61b3 <unknown>\nE       #17 0x555555ce7033 <unknown>\nE       #18 0x7ffffffa6609 start_thread\n\n../venv/lib/python3.9/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException"}, "description": "Random Scenario three", "steps": [{"name": "Given I want to do something three", "status": "failed", "statusDetails": {"message": "Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[text()='Documentation']\"}\n  (Session info: chrome=119.0.6045.10); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n#0 0x555555ce7ea3 <unknown>\n#1 0x5555559aaf97 <unknown>\n#2 0x5555559f8d53 <unknown>\n#3 0x5555559f8e51 <unknown>\n#4 0x555555a3e7f4 <unknown>\n#5 0x555555a1fd1d <unknown>\n#6 0x555555a3b96d <unknown>\n#7 0x555555a1fac3 <unknown>\n#8 0x5555559eaee4 <unknown>\n#9 0x5555559ebeae <unknown>\n#10 0x555555cae701 <unknown>\n#11 0x555555cb259e <unknown>\n#12 0x555555c9c635 <unknown>\n#13 0x555555cb31f6 <unknown>\n#14 0x555555c7ff0f <unknown>\n#15 0x555555cd5fe8 <unknown>\n#16 0x555555cd61b3 <unknown>\n#17 0x555555ce7033 <unknown>\n#18 0x7ffffffa6609 start_thread\n", "trace": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[text()='Documentation']\"}\n  (Session info: chrome=119.0.6045.10); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n#0 0x555555ce7ea3 <unknown>\n#1 0x5555559aaf97 <unknown>\n#2 0x5555559f8d53 <unknown>\n#3 0x5555559f8e51 <unknown>\n#4 0x555555a3e7f4 <unknown>\n#5 0x555555a1fd1d <unknown>\n#6 0x555555a3b96d <unknown>\n#7 0x555555a1fac3 <unknown>\n#8 0x5555559eaee4 <unknown>\n#9 0x5555559ebeae <unknown>\n#10 0x555555cae701 <unknown>\n#11 0x555555cb259e <unknown>\n#12 0x555555c9c635 <unknown>\n#13 0x555555cb31f6 <unknown>\n#14 0x555555c7ff0f <unknown>\n#15 0x555555cd5fe8 <unknown>\n#16 0x555555cd61b3 <unknown>\n#17 0x555555ce7033 <unknown>\n#18 0x7ffffffa6609 start_thread\n\n"}, "start": 1696797970676, "stop": 1696797970922}], "attachments": [{"name": "test_random_three.2023-10-08 21:46:10.922480", "source": "ef881711-91e1-48ec-9bd4-3557e683d778-attachment.png", "type": "image/png"}, {"name": "log", "source": "7163ffb2-2525-4acd-b36c-1d5cd31e2f72-attachment.txt", "type": "text/plain"}], "start": 1696797970676, "stop": 1696797970923, "uuid": "8a1a7a23-eb58-bf18-7479-2fb205993a65", "historyId": "8a1a7a23eb58bf1874792fb205993a65", "fullName": "tests/features/third_feature.feature:Random Scenario three", "labels": [{"name": "host", "value": "GDCMB-104.ultrahub"}, {"name": "thread", "value": "67930-MainThread"}, {"name": "framework", "value": "pytest-bdd"}, {"name": "language", "value": "cpython3"}, {"name": "feature", "value": "Sample feature three"}, {"name": "tag", "value": "usefixtures"}, {"name": "tag", "value": "2108"}], "links": [{"type": "tms", "url": "https://gdcgroup.testrail.io/index.php?/cases/view/2108", "name": "https://gdcgroup.testrail.io/index.php?/cases/view/2108"}]}